storage/connectors/jdbc/src/main/java/feast/storage/connectors/jdbc/retriever/AbstractJdbcQueryTemplater.java

```diff
@@ -0,0 +1,438 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.storage.connectors.jdbc.retriever;
+
+import com.google.protobuf.Duration;
+import com.mitchellbosecke.pebble.PebbleEngine;
+import com.mitchellbosecke.pebble.template.PebbleTemplate;
+import feast.proto.core.FeatureSetProto;
+import feast.proto.serving.ServingAPIProto;
+import feast.storage.api.retriever.FeatureSetRequest;
+import feast.storage.connectors.jdbc.connection.JdbcConnectionProvider;
+import io.grpc.Status;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.sql.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public abstract class AbstractJdbcQueryTemplater implements JdbcQueryTemplater {
```

Comment:
Why is it abstract? I don't see any abstract methods in the class.

Reply:
we make it an abstract class so that other concrete query_templater classes can extend this abstract class and only override those methods that they needed to.  Otherwise, other concrete classes, such as postgress_query_templater can just use the same implements by extending this abstract class.

Reply:
Yes that is correct. but If abstract keyword is removed, `AbstractJdbcQueryTemplater` is same as `PostgressQueryTemplater`

Reply:
yes, currently `PostgresQueryTemplater` doesn't need to override any methods, because the AbstractJdbcQueryTemplater is based on the Postgres template.

Reply:
the abstract class shouldn't refer to any specific implementation.